jsonmodels/validators.py

```diff
@@ -81,41 +81,46 @@ class Regex(object):
 
     """Validator for regular expressions."""
 
-    FLAGS = {
+    ECMA_FLAGS = {
         'ignorecase': re.I,
         'multiline': re.M,
     }
 
-    def __init__(self, pattern, **flags):
+    def __init__(self, pattern, ecma=True, **flags):
         """Init.
 
         Note, that if given pattern is ECMA regex, given flags will be
         **completely ignored** and taken from given regex.
 
 
         :param string pattern: Pattern of regex.
+        :param boolean ecma: Whether or not use ECMA regex, default True.
         :param dict flags: Allowed flags can be found in attribute
-            `ATTRIBUTES_TO_FLAGS`. Invalid flags will be ignored.
+            `ECMA_FLAGS`. Invalid flags will be ignored. Will be ignored if not
+            ECMA regex.
 
         """
-        flags = dict(
-            (key, value) for key, value in flags.items()
-            if key in self.FLAGS
-        )
-
-        if utilities.is_ecma_regex(pattern):
-            result = utilities.convert_ecma_regex_to_python(pattern)
-            self.pattern = result.regex
-
-            for key, _ in flags.items():
-                flags.update(
-                    {key: self.FLAGS[key] in result.flags})
-        else:
-            self.pattern = pattern
+        self.ecma = ecma
+        self.pattern = pattern
+        self.flags = []
+
+        if ecma:
+            flags = dict(
+                (key, value) for key, value in flags.items()
+                if key in self.ECMA_FLAGS
+            )
+
+            if utilities.is_ecma_regex(pattern):
+                result = utilities.convert_ecma_regex_to_python(pattern)
+                self.pattern = result.regex
 
-        self.flags = [
-            self.FLAGS[key] for key, value
-            in flags.items() if value]
+                for key, _ in flags.items():
+                    flags.update(
+                        {key: self.ECMA_FLAGS[key] in result.flags})
+
+            self.flags = [
```

Comment:
This breaks compatibility with current code, and contradicts the comment at the head of the function:
> Note, that if given pattern is ECMA regex, given flags will be
**completely ignored** and taken from given regex.

Reply:
Well, it appears that actually the **code comment is incorrect** :(

Ignore my comment.